### PR TITLE
Global settings for HTTP client 4454

### PR DESCRIPTION
**Pull request by ikapelyukhin**

### CODE REVIEW

**Comment by suse-tests-pass** [comment]
@fschueller will review your pull request :dancers: check https://trello.com/c/rQvPvfjt/4454-5-rmt-make-it-possible-to-work-behind-a-proxy-with-ntlm-auth
  @fschueller: Please review this pull request using our guidelines:
  * test for acceptance criteria / functionality
  * check if the new code is covered with tests
  * check for unintended consequences
  * encourage usage of the boyscout rule
  * make sure the code is architected in the best way
  * check that no unnecessary technical debt got introduced
  * make sure that no unnecessary FIXMEs or TODOs got added

File: spec/lib/rmt/http_request_spec.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      proxy_auth: :ntlm,
+      proxy_user: 'login',
+      proxy_password: 'password'
+    }
+    options.each {|key, value| Settings.http_client.send("#{key}=", value) }
+    options
+  end
+  let(:request) { described_class.new('http://example.com') }
+
+  before { options }
```
Comment (lagartoflojo) [inline spec/lib/rmt/http_request_spec.rb]: Instead of this you can do `let!(:options)`

File: bin/rmt-scc-sync [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+require 'rmt/scc_sync'
+require 'active_record'
+
+require_relative '../app/models/application_record.rb'
+
+Dir.glob(File.join(File.dirname(__FILE__), '../app/models')).each do |folder|
+  Dir.glob(folder + '/*.rb').each do |file|
+    require File.expand_path(file)
+  end
+end
```
Comment (lagartoflojo) [inline bin/rmt-scc-sync]:
```ruby
Dir[File.join(__dir__, '..', 'app', 'models', '**', '*.rb')].each { |file| require File.expand_path(file) }
```

File: bin/rmt-scc-sync [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+db_config_path = File.join(File.dirname(__FILE__), '../config/database.yml')
+db_config = YAML.safe_load(ERB.new(File.read(db_config_path)).result)
```
Comment (lagartoflojo) [inline bin/rmt-scc-sync]: This is failing for me with:

```
/home/hernan/.rvm/rubies/ruby-2.4.1/lib/ruby/2.4.0/psych/visitors/to_ruby.rb:402:in `visit_Psych_Nodes_Alias': Unknown alias: default (Psych::BadAlias)
	from /home/hernan/.rvm/rubies/ruby-2.4.1/lib/ruby/2.4.0/psych/visitors/visitor.rb:16:in `visit'
	from /home/hernan/.rvm/rubies/ruby-2.4.1/lib/ruby/2.4.0/psych/visitors/visitor.rb:6:in `accept'
	from /home/hernan/.rvm/rubies/ruby-2.4.1/lib/ruby/2.4.0/psych/visitors/to_ruby.rb:32:in `accept'
	from /home/hernan/.rvm/rubies/ruby-2.4.1/lib/ruby/2.4.0/psych/visitors/to_ruby.rb:338:in `block in revive_hash'
	from /home/hernan/.rvm/rubies/ruby-2.4.1/lib/ruby/2.4.0/psych/visitors/to_ruby.rb:336:in `each'
	from /home/hernan/.rvm/rubies/ruby-2.4.1/lib/ruby/2.4.0/psych/visitors/to_ruby.rb:336:in `each_slice'
	from /home/hernan/.rvm/rubies/ruby-2.4.1/lib/ruby/2.4.0/psych/visitors/to_ruby.rb:336:in `revive_hash'
	from /home/hernan/.rvm/rubies/ruby-2.4.1/lib/ruby/2.4.0/psych/visitors/to_ruby.rb:162:in `visit_Psych_Nodes_Mapping'
	from /home/hernan/.rvm/rubies/ruby-2.4.1/lib/ruby/2.4.0/psych/visitors/visitor.rb:16:in `visit'
	from /home/hernan/.rvm/rubies/ruby-2.4.1/lib/ruby/2.4.0/psych/visitors/visitor.rb:6:in `accept'
	from /home/hernan/.rvm/rubies/ruby-2.4.1/lib/ruby/2.4.0/psych/visitors/to_ruby.rb:32:in `accept'
	from /home/hernan/.rvm/rubies/ruby-2.4.1/lib/ruby/2.4.0/psych/visitors/to_ruby.rb:338:in `block in revive_hash'
	from /home/hernan/.rvm/rubies/ruby-2.4.1/lib/ruby/2.4.0/psych/visitors/to_ruby.rb:336:in `each'
	from /home/hernan/.rvm/rubies/ruby-2.4.1/lib/ruby/2.4.0/psych/visitors/to_ruby.rb:336:in `each_slice'
	from /home/hernan/.rvm/rubies/ruby-2.4.1/lib/ruby/2.4.0/psych/visitors/to_ruby.rb:336:in `revive_hash'
	from /home/hernan/.rvm/rubies/ruby-2.4.1/lib/ruby/2.4.0/psych/visitors/to_ruby.rb:162:in `visit_Psych_Nodes_Mapping'
	from /home/hernan/.rvm/rubies/ruby-2.4.1/lib/ruby/2.4.0/psych/visitors/visitor.rb:16:in `visit'
	from /home/hernan/.rvm/rubies/ruby-2.4.1/lib/ruby/2.4.0/psych/visitors/visitor.rb:6:in `accept'
	from /home/hernan/.rvm/rubies/ruby-2.4.1/lib/ruby/2.4.0/psych/visitors/to_ruby.rb:32:in `accept'
	from /home/hernan/.rvm/rubies/ruby-2.4.1/lib/ruby/2.4.0/psych/visitors/to_ruby.rb:311:in `visit_Psych_Nodes_Document'
	from /home/hernan/.rvm/rubies/ruby-2.4.1/lib/ruby/2.4.0/psych/visitors/visitor.rb:16:in `visit'
	from /home/hernan/.rvm/rubies/ruby-2.4.1/lib/ruby/2.4.0/psych/visitors/visitor.rb:6:in `accept'
	from /home/hernan/.rvm/rubies/ruby-2.4.1/lib/ruby/2.4.0/psych/visitors/to_ruby.rb:32:in `accept'
	from /home/hernan/.rvm/rubies/ruby-2.4.1/lib/ruby/2.4.0/psych.rb:302:in `safe_load'
	from ./rmt-scc-sync:19:in `<main>'
```

File: bin/rmt-mirror [hunk 4]
```diff
@@ -1,13 +1,11 @@
 #!/usr/bin/env ruby
 
 require_relative '../config/boot'
-require 'pathname'
-require 'config'
+$LOAD_PATH.unshift File.expand_path(File.join(File.dirname(__FILE__), '../lib/'))
```
Comment (lagartoflojo) [inline bin/rmt-mirror]:
```ruby
__dir__ == File.dirname(__FILE__)
```

File: spec/lib/rmt/http_request_spec.rb (hunk 1 above)
Comment (ikapelyukhin) [inline spec/lib/rmt/http_request_spec.rb]: Right! :+1:

File: bin/rmt-scc-sync [hunk 5]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env ruby
+
+require_relative '../config/boot'
+$LOAD_PATH.unshift File.expand_path(File.join(File.dirname(__FILE__), '../lib/'))
```
Comment (lagartoflojo) [inline bin/rmt-scc-sync]: `__dir__`

File: bin/rmt-scc-sync [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+require 'rmt'
+require 'rmt/scc_sync'
+require 'active_record'
+
+require_relative '../app/models/application_record.rb'
+
+Dir[File.join(__dir__, '..', 'app', 'models', '**', '*.rb')].each {|file| require File.expand_path(file) }
+
+db_config_path = File.join(File.dirname(__FILE__), '../config/database.yml')
```
Comment (lagartoflojo) [inline bin/rmt-scc-sync]: `__dir__`

**Pull request merged**